lib/uniffi-zcash/src/orchard/address.rs

```diff
@@ -1,12 +1,33 @@
+use std::sync::Arc;
+
 use orchard::Address;
 
+use crate::{utils, ZcashError, ZcashOrchardDiversifier, ZcashResult};
+
 /// A shielded payment address.
 pub struct ZcashOrchardAddress(pub Address);
 
 impl ZcashOrchardAddress {
+    /// Returns the [`Diversifier`] for this `Address`.
+    pub fn diversifier(&self) -> Arc<ZcashOrchardDiversifier> {
+        Arc::new(self.0.diversifier().into())
+    }
+
+    /// Serializes this address to its "raw" encoding as specified in [Zcash Protocol Spec § 5.6.4.2: Orchard Raw Payment Addresses][orchardpaymentaddrencoding]
+    ///
+    /// [orchardpaymentaddrencoding]: https://zips.z.cash/protocol/protocol.pdf#orchardpaymentaddrencoding
     pub fn to_raw_address_bytes(&self) -> Vec<u8> {
         self.0.to_raw_address_bytes().to_vec()
     }
+
+    /// Parse an address from its "raw" encoding as specified in [Zcash Protocol Spec § 5.6.4.2: Orchard Raw Payment Addresses][orchardpaymentaddrencoding]
+    ///
+    /// [orchardpaymentaddrencoding]: https://zips.z.cash/protocol/protocol.pdf#orchardpaymentaddrencoding
+    pub fn from_raw_address_bytes(bytes: Vec<u8>) -> ZcashResult<Self> {
+        let array = utils::cast_slice(&bytes)?;
+        let address: Option<_> = Address::from_raw_address_bytes(&array).into();
+        Ok(ZcashOrchardAddress(address.ok_or(ZcashError::Unknown)?))
```

Comment:
nitpick: Should we return a more user friendly error here ? I am thinking in a simple string like "error parsing address".

Reply:
Done.